lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -140,15 +144,39 @@ sub _parse_obs_response_dirty {
     return (0, 'published');
 }
 
+sub _get_batches {
+    my ($c, $project, $only_first) = @_;
+    my $home = $c->obs_rsync->home;
+
+    my $batches = Mojo::File->new($home, $project)->list({dir => 1})->grep(sub { -d $_ })->map('basename');
+
+    if ($only_first) {
+        return "" if !$batches->size;
+        my $arr = $batches->to_array();
+        return @$arr[0];
+    }
+    return $batches->to_array();
```

Comment:
It would look much better to write this as `return $batches->to_array unless $only_first` and move it before the `if` block which would then just go away.;